src/python/TaskWorker/Actions/DBSDataDiscovery.py

```diff
@@ -483,6 +491,10 @@ def executeInternal(self, *args, **kwargs):
                 self.checkBlocksSize(secondaryBlocksWithLocation)
         try:
             filedetails = self.dbs.listDatasetFileDetails(inputDataset, getParents=True, getLumis=needLumiInfo, validFileOnly=0)
+            if inputBlocks:
```

Comment:
while this works, it would be good to make a note to optimize load on DBS a little bit and only retrieve info for the block(s) we need instead of querying a possibly x100 bigger list and throw away 99% of it. This may require to use directly DBS API's rather than the WMCore wrapper which we call here. Getting read of that wrapper would be a good things anyhow, but ... again.. an optimization which we can postpone (indefinetly ?)

Reply:
Lets me try. 
I will give up if I cannot finish it next Monday. Then, we will postpone until next time we have problem :)

Reply:
After one night of thinking, I changed my mind. We can do "optimization" later.

We already fetch all file info for all tasks TW process, and we have never had a problem so far. Adding lines of code to maintain in exchange for unknown performance improvement is not a good deal to me.

Note that if we want to fetch blocks. It is easy enough to do because DbsApi [listFileArray()](https://github.com/dmwm/DBSClient/blob/28a1fd60578031e3422eeea8afb8b6bf5d3e6036/src/python/dbs/apis/dbsClient.py#L1474) already supports fetch by block.